travis/self_tests

```diff
@@ -1,77 +1,577 @@
+            u"\033[1;32m\033[0;m\n\033[1;32mtest\033"
+            "[0;m\n\033[1;32mnewline\033[0;m")
+        self.assertEqual(
+            travis_helpers.yellow(u'\ntest\nnewline'),
+            u"\033[1;33m\033[0;m\n\033[1;33mtest\033"
+            "[0;m\n\033[1;33mnewline\033[0;m")
+        self.assertEqual(
+            travis_helpers.yellow_light(u'\ntest\nnewline'),
+            u"\033[33m\033[0;m\n\033[33mtest\033"
+            "[0;m\n\033[33mnewline\033[0;m")
+
+    def test_03(self):
+        """Testing empty paths and pylint_run fix of:
+           https://www.mail-archive.com/code-quality@python.org/msg00294.html
+        """
+        if os.environ.get('LINT_CHECK', 0) == '1':
+            print "CHECK PYLINT MATRIX VERSIONS"
+            self.check_vmaster()
+            self.check_v7()
+            self.check_v8()
+            self.check_without_version()
+
+            print "Standard Check"
+            pylint_rcfile = os.path.join(
+                os.path.dirname(os.path.realpath(__file__)),
+                'cfg', "travis_run_pylint.cfg")
+            count_errors = run_pylint.main([
+                "--config-file=" + pylint_rcfile,
+                "--extra-params", "-d", "--extra-params",
+                "all", "--extra-params", "-e",
+                "--extra-params", "F0010,duplicate-key",
+                "--path", self.repo_dir], standalone_mode=False)
+            self.assertEqual(2, count_errors)
+
+            empty_path = os.path.join(self.repo_dir, 'empty_path')
+            if not os.path.exists(empty_path):
+                os.mkdir(empty_path)
+            count_errors = run_pylint.main([
+                "--config-file=" + pylint_rcfile,
+                "--path", empty_path], standalone_mode=False)
+            self.assertEqual(-1, count_errors)
+            if os.environ.get('TRAVIS_PULL_REQUEST', 'false') == 'true':
+                git_script_path = os.path.join(os.path.dirname(
+                    os.path.dirname(os.path.realpath(__file__))), 'git')
+                pre_commit_returned = subprocess.call(os.path.join(
+                    git_script_path, 'pre-commit'))
+                assert pre_commit_returned == 0, \
+                    "Git pre-commit script returned value != 0"
+
+    def test_04(self):
+        """Testing git get branch"""
+        pass
+
+    def test_05(self):
+        """Testing git run from getaddons"""
+        getaddons.get_modules_changed(self.repo_dir)
+
+    def check_vmaster(self):
+        """PYLINT_EXPECTED_ERRORS="89"
+           VERSION="master"
+        """
+        print "VERSION=MASTER"
+        expected_errors = dict(EXPECTED_ERRORS)
+        self.pylint_run(
+            is_pr=True, version="master",
+            expected_errors=91, errors_dict=expected_errors)
+
+        expected_errors = dict(EXPECTED_ERRORS)
+        expected_errors.pop('openerp-exception-warning')
+        expected_errors.pop('redundant-modulename-xml')
```

Comment:
What about use a global variable for `expected_errors` and other one for `expected_errors_pr`?
In order to avoid `pop` many times in many methods.

And you could use: ` self.pylint_run(is_pr=True, ..., errors_dict=expected_errors + expected_errors_pr)`